core/chains/evm/forwarders/forwarder_manager.go

```
@@ -0,0 +1,308 @@
+
+var authLogCheckInterval = time.Duration(5 * time.Second)
+
+type FwdMgr struct {
+	ORM       ORM
+	config    Config
+	evmClient evmclient.Client
+	logger    logger.SugaredLogger
+	logpoller *evmlogpoller.LogPoller
+
+	// TODO(samhassan): sendersCache should be an LRU capped cache
+	sendersCache map[common.Address][]common.Address
+	latestBlock  int64
+
+	authRcvr    authorized_receiver.AuthorizedReceiverInterface
+	offchainAgg offchain_aggregator_wrapper.OffchainAggregatorInterface
+
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	cacheMu sync.RWMutex
+	chStop  chan struct{}
+	wg      sync.WaitGroup
+	utils.StartStopOnce
+}
+
+func NewFwdMgr(db *sqlx.DB, cfg Config, client evmclient.Client, logpoller *evmlogpoller.LogPoller, l logger.Logger) *FwdMgr {
+	lggr := logger.Sugared(l.Named("FwdMgr"))
+	fwdMgr := FwdMgr{
+		logger:       lggr,
+		evmClient:    client,
+		ORM:          NewORM(db, lggr, cfg),
+		logpoller:    logpoller,
+		config:       cfg,
+		sendersCache: make(map[common.Address][]common.Address),
+		chStop:       make(chan struct{}),
+		cacheMu:      sync.RWMutex{},
+		wg:           sync.WaitGroup{},
+		latestBlock:  0,
+	}
+	return &fwdMgr
+}
+
+// Start starts Forwarder Manager.
+func (f *FwdMgr) Start() error {
+	return f.StartOnce("EVMForwarderManager", func() error {
+		f.logger.Debug("Initializing EVM forwarder manager")
+
+		fwdrs, cnt, err := f.ORM.FindForwardersByChain(utils.Big(*f.evmClient.ChainID()))
+		if err != nil {
+			return errors.Wrapf(err, "Failed to retrieve forwarders for chain %d", f.evmClient.ChainID())
+		}
+		f.ctx, f.cancel = context.WithCancel(context.Background())
+		if cnt != 0 {
+			f.initForwardersCache(f.ctx, fwdrs)
+			f.subscribeForwardersLogs(fwdrs)
+		}
+
+		f.authRcvr, err = authorized_receiver.NewAuthorizedReceiver(common.Address{}, f.evmClient)
+		if err != nil {
+			f.logger.Panic("Failed to init AuthorizedReceiver")
```

do we want to return an error in this case?

---

error seems more graceful. Done